trawler.py

```diff
@@ -0,0 +1,88 @@
+from selenium import webdriver
+from urllib.parse import urlparse, urljoin
+import sys
+import re
+
+class Scraper():
+    def __init__(self):
+        self.s = webdriver.Chrome()
+
+    def go(self, url):
+        self.s.get(url)
+        return self.s.page_source
+
+# Filter out various downloadable files that probably don't have emails in
+# metadata. regex is probably overkill and can lead to poor performance,
+# but network io is likely a larger bottleneck.
+def is_web_page(url):
+    return not re.search('(?i)\.(?:jpe?g|tiff?|bmp|rar|gif|png|zip|torrent|gz|bz2|gzip|docx?|xlsx?|pptx?|pdf|tex|dvi|ps|au|tar|ics)$', url)
```

Comment:
I think it might make more sense to check the MIME type (skipping everything that isn't text, or possibly explicitly text/html), although I'm not sure if that's doable with Selenium.